# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/websocket/WebSocket.cpp
# Commit: 379cd5044768
# Full Hash: 379cd5044768c61ee7b94cc383fa498433a30999
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P1 Make WorkerRunnable to be a base class for runnables on Worker thread and Worker's parent thread. r=dom-worker-reviewers,asuth
#   
#   This is the first step in splitting the parent thread runnable out of WorkerRunnable.
#   
#   To reuse the runnable dispatching codes in Worker, we still need a base class for runnable on the worker thread and the parent thread.
# ==============================================================================

diff -r 29cb10d35b5a -r 379cd5044768 dom/websocket/WebSocket.cpp
--- a/dom/websocket/WebSocket.cpp	Fri Apr 19 09:32:48 2024 +0000
+++ b/dom/websocket/WebSocket.cpp	Fri Apr 19 09:41:57 2024 +0000
@@ -2785,15 +2785,15 @@
 
 namespace {
 
-class WorkerRunnableDispatcher final : public WorkerRunnable {
+class WorkerRunnableDispatcher final : public WorkerThreadRunnable {
   RefPtr<WebSocketImpl> mWebSocketImpl;
 
  public:
   WorkerRunnableDispatcher(WebSocketImpl* aImpl,
                            ThreadSafeWorkerRef* aWorkerRef,
                            already_AddRefed<nsIRunnable> aEvent)
-      : WorkerRunnable(aWorkerRef->Private(), "WorkerRunnableDispatcher",
-                       WorkerThread),
+      : WorkerThreadRunnable(aWorkerRef->Private(), "WorkerRunnableDispatcher",
+                             WorkerRunnable::WorkerThread),
         mWebSocketImpl(aImpl),
         mEvent(std::move(aEvent)) {}
 